instr.ml

```diff
@@ -129,6 +129,7 @@ type analysis_input = {
   formals : VarSet.t;
   instrs : instructions;
 }
+type analysis = analysis_input -> instructions option
```

Comment:
i find `analysis` confusing. For me `analysis` is `analysis_input -> analysis_result` whereas this one is a transformation (ie. an analysis just looks at the code whereas a transformation creates new code).